common/core/desktop/tests/unit/kmx/kmx_key_list.cpp

```diff
@@ -0,0 +1,86 @@
+/*
+  Copyright:    © 2018 SIL International.
+  Description:  Tests for key rules list generation in the kmx processor
+  Create Date:  30 Oct 2018
+  Authors:      Ross Cruickshank (RC) Marc Durdin (MD), Tim Eves (TSE)
+
+*/
+
+#include <kmx/kmx_processevent.h>
+
+#include "path.hpp"
+#include "state.hpp"
+
+#include "../test_assert.h"
+#include "../test_color.h"
+
+using namespace km::kbp::kmx;
+
+km_kbp_option_item test_env_opts[] =
+{
+  KM_KBP_OPTIONS_END
+};
+
+int error_args() {
+    std::cerr << "kmx: Not enough arguments." << std::endl;
+    return 1;
+}
+
+km_kbp_keyboard_key kb_key_expected_list[] = {{49,16384},{50,16384},{66,16416},{67,16384},{67,16448},{97,0},{98,0}};
```

Comment:
The test might be easier to understand if we'd use the `KM_KBP_VKEY_*` and `KM_KBP_MODIFIER_*` consts here, especially when looking at the .kmn file and trying to understand the expected list here.